From 2995ef8e46934b76d30ed0b5fd9480f319e75308 Mon Sep 17 00:00:00 2001
From: Serhey Popovych <serhe.popovych@gmail.com>
Date: Fri, 24 Feb 2017 11:08:55 +0200
Subject: [PATCH 05/11] veth: Set ifindex only if given and not negative

There is already check for ifindex being non negative in
register_netdevice(). Do the same here for convenience.

Fixes: e6f8f1a739b6 ("veth: Allow to create peer link with given ifindex")
Signed-off-by: Serhey Popovych <serhe.popovych@gmail.com>
---
 drivers/net/veth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/veth.c b/drivers/net/veth.c
index 5be8ed9..bdeb2a5 100644
--- a/drivers/net/veth.c
+++ b/drivers/net/veth.c
@@ -1351,7 +1351,7 @@ static int veth_newlink(struct net *src_net, struct net_device *dev,
 	if (!ifmp || !tbp[IFLA_ADDRESS])
 		eth_hw_addr_random(peer);
 
-	if (ifmp && (dev->ifindex != 0))
+	if (ifmp && dev->ifindex > 0)
 		peer->ifindex = ifmp->ifi_index;
 
 	peer->gso_max_size = dev->gso_max_size;
-- 
1.8.3.1

